### ISSUE TITLE
Bug: `no-useless-computed-key` does not report computed keys using backticks

### ISSUE THREAD

**Issue by abrahamguo** [eslint/eslint#21043]
### Environment

Local ESLint version: latest (10.6.0)

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

```js
{ rules: { 'no-useless-computed-key': 'error' } }
```
</details>

```js
({ [`a`]: 1, ['b']: 2 });
```

### What did you expect to happen?

The rule should flag both of the two useless computed keys.

### What actually happened?

The rule does not flag the first key because it is a `TemplateLiteral` rather than a `Literal`.

### Link to Minimal Reproducible Example

https://eslint.org/play/#eyJ0ZXh0IjoiKHsgW2BhYF06IDEsIFsnYiddOiAyIH0pOyIsIm9wdGlvbnMiOnsicnVsZXMiOnsibm8tdXNlbGVzcy1jb21wdXRlZC1rZXkiOlsiZXJyb3IiXX0sImxhbmd1YWdlT3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFGZWF0dXJlcyI6e319fX19

### Participation

- [x] I am willing to submit a pull request for this issue.

### AI acknowledgment

- [x] I did not use AI to generate this issue report.
- [ ] (If the above is not checked) I have reviewed the AI-generated content before submitting.

### Additional comments

  I am aware that the rule is frozen; however, per [ESLint's documentation on frozen rules](https://eslint.org/docs/latest/contribute/core-rules#frozen-rules), ESLint will still ship bug fixes. I would classify this as a bug.

**Comment by mdjermanovic** [comment]
This is intentional behavior because removing `[]` would produce invalid syntax (please see https://github.com/eslint/eslint/issues/19280).
